CONTRIBUTING.rst

```diff
@@ -6,12 +6,29 @@ Where possible include updates to NEWS along with your improvements.
 
 New functionality and bug fixes should be accompanied by matching unit tests.
 
+Installing development dependencies
+-----------------------------------
+
+Install the checked out Dulwich package in editable mode with ``dev`` extras:
+
+.. code:: console
+
+   $ cd ~/path/to/checkouts/dulwich
+   # Make sure to run this in a virtual environment
+   $ pip install -e ".[dev]"
```

Comment:
This seems a little prescriptive - I've personally never used -e.

Reply:
Sure, I can make it less prescriptive and more of a "here's an easy way to start".

Reply:
I adjusted the wording to make it more of general outline, and expanded it to mention Rust / Cargo being required as well.